src/test/clojure/clara/test_java_facts.clj

```diff
@@ -0,0 +1,29 @@
+(ns clara.test-java-facts
+  (:require [clara.tools.testing-utils :as tu]
+            [clara.rules :as rules]
+            [clojure.test :refer [is deftest run-tests testing use-fixtures]])
+  (:import [clara.test.facts
+            BeanTestFact]))
+
```

Comment:
This test seems like it could go in the test_java namespace as well, but I don't have strong feelings about it.

Reply:
I was on the fence about that, I kind of felt like the test_java namespace was testing the Java apis rather than the usage of java roots in the rulebase and thats why i created a new namespace.